In [ ]:
from src.googleapi import *

In [2]:
place = find_aggregated_places()


! ERROR: 429


In [3]:
print(place['error']['message'])

The response exceeded the maximum number of allowed places. Only 100 places can be returned in a single response. The request produced 2648 places and therefore no places were returned.


> The response exceeded the maximum number of allowed places. Only 100 places can be returned in a single response. The request produced **2648** places and therefore no places were returned.

# Bounding box

coordinates for copenhagen in a bounding box:

(topleft)
55.82 12.23 
(bottomright)
55.51 12.73

In [26]:
# checking with aggregated places
import googleapi

COPENHAGEN_BOUNDS = {
    "lat_south": 55.51,
    "lat_north": 55.82,
    "lon_west": 12.23,
    "lon_east": 12.73,
}

results = googleapi.find_aggregated_places(COPENHAGEN_BOUNDS)
    
print(results['error']['message'])

Elapsed time: 999.92975 milliseconds for thread MAKE_REQUEST(URL='HTTPS://AREAINSIGHTS.GOOGLEAPIS.COM/V1:COMPUTEINSIGHTS', BODY={'INSIGHTS': ['INSIGHT_COUNT', 'INSIGHT_PLACES'], 'FILTER': {'LOCATIONFILTER': {'CUSTOMAREA': {'POLYGON': {'COORDINATES': [{'LATITUDE': 55.82, 'LONGITUDE': 12.23}, {'LATITUDE': 55.51, 'LONGITUDE': 12.23}, {'LATITUDE': 55.51, 'LONGITUDE': 12.73}, {'LATITUDE': 55.82, 'LONGITUDE': 12.73}, {'LATITUDE': 55.82, 'LONGITUDE': 12.23}]}}}, 'TYPEFILTER': {'INCLUDEDTYPES': 'RESTAURANT'}}})
Elapsed time: 1.00017 seconds for thread FIND_AGGREGATED_PLACES(COORDINATES={'LAT_SOUTH': 55.51, 'LAT_NORTH': 55.82, 'LON_WEST': 12.23, 'LON_EAST': 12.73})
The response exceeded the maximum number of allowed places. Only 100 places can be returned in a single response. The request produced 4218 places and therefore no places were returned.


Sometimes returns 4218 and sometimes returns 3968 places. Why?

In [27]:
path = '../data/copenhagen-bounds_20260312_115801.json'

In [28]:
import pandas as pd

df = pd.read_json(path)

In [29]:
df.shape

(4364, 66)

In [30]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 4364 entries, 0 to 4363
Data columns (total 66 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   name                          4364 non-null   str    
 1   id                            4364 non-null   str    
 2   types                         4364 non-null   object 
 3   nationalPhoneNumber           3748 non-null   str    
 4   internationalPhoneNumber      3748 non-null   str    
 5   formattedAddress              4364 non-null   str    
 6   addressComponents             4364 non-null   object 
 7   plusCode                      4353 non-null   object 
 8   location                      4364 non-null   object 
 9   viewport                      4364 non-null   object 
 10  rating                        4017 non-null   float64
 11  googleMapsUri                 4364 non-null   str    
 12  websiteUri                    3590 non-null   str    
 13  regularOpening

In [31]:
pd.set_option("display.max.columns", None)

In [32]:
df.head()

,name,id,types,nationalPhoneNumber,internationalPhoneNumber,formattedAddress,addressComponents,plusCode,location,viewport,rating,googleMapsUri,websiteUri,regularOpeningHours,utcOffsetMinutes,adrFormatAddress,businessStatus,priceLevel,userRatingCount,iconMaskBaseUri,iconBackgroundColor,displayName,primaryTypeDisplayName,takeout,delivery,dineIn,reservable,servesBreakfast,servesLunch,servesDinner,servesBeer,servesWine,servesBrunch,currentOpeningHours,currentSecondaryOpeningHours,regularSecondaryOpeningHours,primaryType,shortFormattedAddress,reviews,photos,outdoorSeating,liveMusic,menuForChildren,servesCocktails,servesDessert,servesCoffee,goodForChildren,allowsDogs,restroom,goodForGroups,goodForWatchingSports,paymentOptions,parkingOptions,accessibilityOptions,addressDescriptor,googleMapsLinks,priceRange,timeZone,postalAddress,googleMapsTypeLabel,curbsidePickup,containingPlaces,editorialSummary,fuelOptions,servesVegetarianFood,consumerAlert
0,places/ChIJ6eXKTWdDUkYR7k5XjV6Ju-0,ChIJ6eXKTWdDUkYR7k5XjV6Ju-0,"[restaurant, inn, lodging, point_of_interest, ...",48 18 30 01,+45 48 18 30 01,"Bygaden 30, 3660 Stenløse, Denmark","[{'longText': '30', 'shortText': '30', 'types'...","{'globalCode': '9F7JQ7V7+94', 'compoundCode': ...","{'latitude': 55.793493999999995, 'longitude': ...","{'low': {'latitude': 55.792122919708504, 'long...",4.4,https://maps.google.com/?cid=17130436646825053...,http://ganlosekro.dk/,"{'openNow': False, 'periods': [{'open': {'day'...",60,"<span class=""street-address"">Bygaden 30</span>...",OPERATIONAL,PRICE_LEVEL_MODERATE,1376.0,https://maps.gstatic.com/mapfiles/place_api/ic...,#FF9E67,"{'text': 'Ganløse Kro', 'languageCode': 'da'}","{'text': 'Restaurant', 'languageCode': 'en-US'}",1.0,0.0,1.0,1.0,0.0,1.0,1.0,1.0,1.0,1.0,"{'openNow': False, 'periods': [{'open': {'day'...","[{'openNow': False, 'periods': [{'open': {'day...","[{'openNow': False, 'periods': [{'open': {'day...",restaurant,"Bygaden 30, Stenløse",[{'name': 'places/ChIJ6eXKTWdDUkYR7k5XjV6Ju-0/...,[{'name': 'places/ChIJ6eXKTWdDUkYR7k5XjV6Ju-0/...,0.0,1.0,1.0,1.0,1.0,1.0,1.0,0.0,1.0,1.0,0.0,"{'acceptsCreditCards': True, 'acceptsDebitCard...","{'freeParkingLot': True, 'freeStreetParking': ...","{'wheelchairAccessibleParking': True, 'wheelch...",{'landmarks': [{'name': 'places/ChIJdYNxqmBDUk...,{'directionsUri': 'https://www.google.com/maps...,"{'startPrice': {'currencyCode': 'DKK', 'units'...",{'id': 'Europe/Copenhagen'},"{'regionCode': 'DK', 'languageCode': 'en-US', ...","{'text': 'Restaurant', 'languageCode': 'en-US'}",NaN,NaN,NaN,NaN,NaN,NaN
1,places/ChIJ21qBwtRDUkYRGBAxnmFfg10,ChIJ21qBwtRDUkYRGBAxnmFfg10,"[restaurant, point_of_interest, food, establis...",93 92 21 42,+45 93 92 21 42,"Bygaden 7, 3660 Stenløse, Denmark","[{'longText': '7', 'shortText': '7', 'types': ...","{'globalCode': '9F7JQ7R7+C8', 'compoundCode': ...","{'latitude': 55.7911146, 'longitude': 12.263354}","{'low': {'latitude': 55.789767219708494, 'long...",4.7,https://maps.google.com/?cid=67383343403468759...,http://www.hjaltesburesoe.dk/,"{'openNow': True, 'periods': [{'open': {'day':...",60,"<span class=""street-address"">Bygaden 7</span>,...",OPERATIONAL,PRICE_LEVEL_MODERATE,139.0,https://maps.gstatic.com/mapfiles/place_api/ic...,#FF9E67,"{'text': 'Hjaltes Ganløse', 'languageCode': 'da'}","{'text': 'Restaurant', 'languageCode': 'en-US'}",1.0,0.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,1.0,"{'openNow': True, 'periods': [{'open': {'day':...",NaN,NaN,restaurant,"Bygaden 7, Stenløse",[{'name': 'places/ChIJ21qBwtRDUkYRGBAxnmFfg10/...,[{'name': 'places/ChIJ21qBwtRDUkYRGBAxnmFfg10/...,1.0,0.0,1.0,NaN,1.0,1.0,1.0,NaN,1.0,1.0,0.0,"{'acceptsCreditCards': True, 'acceptsDebitCard...","{'freeParkingLot': True, 'freeStreetParking': ...",NaN,{'landmarks': [{'name': 'places/ChIJm0zBFl5DUk...,{'directionsUri': 'https://www.google.com/maps...,"{'startPrice': {'currencyCode': 'DKK', 'units'...",{'id': 'Europe/Copenhagen'},"{'regionCode': 'DK', 'languageCode': 'en-US', ...","{'text': 'Brasserie', 'languageCode': 'en-US'